File: back/src/main/java/com/back/domain/chat/chatRoom/dto/ChatRoomDto.java
package com.back.domain.chat.chatRoom.dto;

import com.back.domain.chat.chatRoom.entity.ChatRoom;
import com.back.domain.chat.chatRoom.entity.ChatRoomStatus;
import com.back.domain.match.matchRequest.entity.Situation;
import com.fasterxml.jackson.annotation.JsonInclude;
import java.time.LocalDateTime;
import java.util.UUID;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record ChatRoomDto(
        UUID roomId,
        ChatRoomStatus status,
        int maxParticipants,
        LocalDateTime createdAt,
        LocalDateTime closedAt,
        boolean isBot,
        Situation opponentSituation
) {
    public ChatRoomDto(ChatRoom chatRoom, boolean isBot, Situation opponentSituation) {
        this(
                chatRoom.getId(),
                chatRoom.getStatus(),
                chatRoom.getMaxParticipants(),
                chatRoom.getCreatedAt(),
                chatRoom.getClosedAt(),
                isBot,
                opponentSituation
        );
    }
}

File: back/src/main/kotlin/com/back/domain/chat/chatRoom/dto/ChatRoomDto.kt
package com.back.domain.chat.chatRoom.dto

import com.back.domain.chat.chatRoom.entity.ChatRoom
import com.back.domain.chat.chatRoom.entity.ChatRoomStatus
import com.back.domain.match.matchRequest.entity.Situation
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime
import java.util.UUID

@JsonInclude(JsonInclude.Include.NON_NULL)
data class ChatRoomDto(
    val roomId: UUID,
    val status: ChatRoomStatus,
    val maxParticipants: Int,
    val createdAt: LocalDateTime,
    val closedAt: LocalDateTime?,
    @get:JsonProperty("isBot") val isBot: Boolean,
    val opponentSituation: Situation?
) {
    constructor(chatRoom: ChatRoom, isBot: Boolean, opponentSituation: Situation?) : this(
        chatRoom.uuid,
        chatRoom.status,
        chatRoom.maxParticipants,
        chatRoom.createdAt,
        chatRoom.closedAt,
        isBot,
        opponentSituation
    )
}
